river/tree/splitter/nominal_class_splitter.py

```diff
@@ -1,16 +1,16 @@
 from .._attribute_test import (
-    AttributeSplitSuggestion,
     NominalAttributeBinaryTest,
     NominalAttributeMultiwayTest,
+    SplitSuggestion,
 )
-from .attribute_observer import AttributeObserver
+from .base_splitter import Splitter
 
 
-class NominalAttributeClassObserver(AttributeObserver):
-    """Class for observing the class data distribution for a nominal attribute.
-    This observer monitors the class distribution of a given attribute.
-    Used in naive Bayes and decision trees to monitor data statistics on leaves.
+class NominalClassSplitter(Splitter):
```

Comment:
The new names are kind of confusing. As they got shorter what about: `NominalSplitterClassif` and `NominalSplitterReg` respectively?

Reply:
Yes! Much better haha

What was I thinking? Nominal Reg/Class Splitter, LOL

Changed in commit [b71f853](https://github.com/online-ml/river/pull/488/commits/b71f853f5379e0e61b1b70902babc2ff3a4f3ce0).